Q1: What is the difference between a t-test and a z-test? Provide an example scenario where you would
use each type of test.

Ans: If we dont know the standard deviation of the population then we will use t-test.But when we know the standard devviation and the number of sample is above 30,then we will use z-test,but if the number of sample will be less than 30 then we will use t-test.

Example: Populaton mean is 520. 40 samples are drawn with sample mean 500 and standard deviation2.1 .
Here thenumbe of sample is greater than 30 i.e. why we will use z-test.
But if the question is ,Populaton mean is 520. 20 samples are drawn with sample mean 500 ,then as no standard deviation is given and the number of sample is less than 30,so we will use t-test .

Q2: Differentiate between one-tailed and two-tailed tests.

Ans: The main difference between one-tailed and two-tailed tests is that one-tailed tests will only have one critical region whereas two-tailed tests will have two critical regions.

Q3: Explain the concept of Type 1 and Type 2 errors in hypothesis testing. Provide an example scenario for
each type of error.

Ans: If we reject null hypothesis when in reality it is true,then we will call it as Type 1 error.
When we retain the null hypothesis when in reality it is false, then we call it as type 2 error.

Example: Populaton mean is 520. 40 samples are drawn with sample mean 500 and standard deviation2.1 . Here thenumbe of sample is greater than 30 .
if we rejct the null hypothesis i.e. population mean=sample mean when it is true then we will call it as type 1 error.
But if we accept the null hypothesis i.e. population mean=sample mean when it is false then we will call it as type 2 error.

Q4: Explain Bayes's theorem with an example.

Ans:Bayes thorem is a mathematical formula for determining conditional probability. Conditional probability is the likelihood of an outcome occurring, based on a previous outcome having occurred in similar circumstances. Bayes' theorem provides a way to revise existing predictions or theories  given new or additional evidence.
the notation is  P(A|B) = P(A)P(B|A)/P(B)
Example: if we have to calculate the probability of taking a blue ball from the second bag out of three different bags of balls, where each bag contains three different colour balls viz. red, blue, black. In this case, the probability of occurrence of an event is calculated depending on other conditions is known as conditional probability. 

Q5: What is a confidence interval? How to calculate the confidence interval, explain with an example.

Ans: confidence interval helps to estimate what the actual value of the unknown population mean is.

confidence interval = point estimate +/- margin of error

if the point estimate is 5 and the margin of error is .5 then the confidence integral will be 4.5 - 5.5. 

Q6. Use Bayes' Theorem to calculate the probability of an event occurring given prior knowledge of the
event's probability and new evidence. Provide a sample problem and solution.

Ans: probability of an event occurring given prior knowledge of the event's probability and new evidence is denoted as P(A|B),the formula is = P(A|B) = P(A)P(B|A)/P(B)

Example:  
<!-- A bag 1 contains 4 white and 6 black balls while another Bag 2 contains 4 white and 3 black balls. One ball is drawn at random from one of the bags, and it is found to be black. Find the probability that it was drawn from Bag 1 -->
solution: Let E1 be the event of choosing bag I, E2 the event of choosing bag II, and A be the event of drawing a black ball.
i.e. P(E1) = P(E2) = 1/2
P(A|E1) = 6/10
P(A|E2) = 3/7
By using Bayes’ theorem, the probability of drawing a black ball from bag I out of two bags,
P(E1|A) = P(E1)P(A|E1)/(P(E1)P(A|E1)+P(E2)P(A|E2)) = 7/12



Q7. Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation
of 5. Interpret the results.

Ans: 
Since 95% of values fall within two standard deviations of the mean according to the 68-95-99.7 Rule, simply add and subtract two standard deviations from the mean in order to obtain the 95% confidence interval.i.e. the the c.i.will be (50-5)=45 to(50+5)=55

Q8. What is the margin of error in a confidence interval? How does sample size affect the margin of error?
Provide an example of a scenario where a larger sample size would result in a smaller margin of error.

Margin of error tells you how much you can expect your survey results to reflect the views from the overall population.
 the larger the sample size, the smaller the sample standard error and therefore the smaller the margin of error.
Average  = 5.5. mins; Standard Deviation = 3 mins; N = 10 runs
95% Confidence Interval = (3.6, 7.4)
Margin of Error = ±1.9 minutes
Average = 5.5 mins; Standard Deviation = 3 mins; N=1000 runs
95% Confidence Interval = (5.4, 5.6)
Margin of Error = ± 0.1 minutes
Thus margin of error decreses when the sample size increases

Q9. Calculate the z-score for a data point with a value of 75, a population mean of 70, and a population
standard deviation of 5. Interpret the results.

In [5]:
pop_mean = 70
pop_std = 53
samp = 75
z_score = (samp-pop_mean)/pop_std
z_score

0.09433962264150944

Q10. In a study of the effectiveness of a new weight loss drug, a sample of 50 participants lost an average
of 6 pounds with a standard deviation of 2.5 pounds. Conduct a hypothesis test to determine if the drug is
significantly effective at a 95% confidence level using a t-test.

In [7]:
import math
n = 50
samp_mean = 6
samp_std = 2.5
ci = .95
alpha = 1-ci
t_value_statistics = -2.010
pop_mean = ((samp_mean-t_value_statistics)*math.sqrt(n-1))/samp_std
t_score = (samp_mean-pop_mean)/(samp_std/math.sqrt(n))

In [12]:
if t_score < t_value_statistics:
    print("We will reject the null hypothesis")
else:
    print("We fail to reject the fail null hypothesis")

We will reject the null hypothesis


Q11. In a survey of 500 people, 65% reported being satisfied with their current job. Calculate the 95%
confidence interval for the true proportion of people who are satisfied with their job.

In [13]:
n = 500
p = .65
q = 1-p
ci = .95
alpha = 1- ci
z_score = 1.96
EBP = z_score*math.sqrt((p*q)/n)

In [15]:
p-EBP

0.608191771144905

In [16]:
p+EBP

0.6918082288550951

i.e. the confidence interval is : 60% -69%

Q12. A researcher is testing the effectiveness of two different teaching methods on student performance.
Sample A has a mean score of 85 with a standard deviation of 6, while sample B has a mean score of 82
with a standard deviation of 5. Conduct a hypothesis test to determine if the two teaching methods have a
significant difference in student performance using a t-test with a significance level of 0.01.

In [17]:
xa_bar = 85
xa_std = 6
xb_bar = 82
xb_std = 5
alpha = 0.01
t = xa_bar-xb_bar




SyntaxError: invalid syntax (3022795054.py, line 6)

Q13. A population has a mean of 60 and a standard deviation of 8. A sample of 50 observations has a mean
of 65. Calculate the 90% confidence interval for the true population mean.

In [22]:
pop_mean = 60
std = 8
samp_mean = 65
n = 50
ci = .90
a = 1-ci
z_score = 1.65
z_test = (samp_mean-pop_mean)/std/math.sqrt(n)
lower_ci = samp_mean - z_test*(std/math.sqrt(n))
upper_ci = samp_mean + z_test*(std/math.sqrt(n))

In [23]:
lower_ci

64.9

In [24]:
upper_ci

65.1

the confidence integral is 64.9 - 65.1

Q14. In a study of the effects of caffeine on reaction time, a sample of 30 participants had an average
reaction time of 0.25 seconds with a standard deviation of 0.05 seconds. Conduct a hypothesis test to
determine if the caffeine has a significant effect on reaction time at a 90% confidence level using a t-test.

In [32]:
n = 30
samp_mean = 0.25
samp_std = 0.05
ci = .90
a= .10
t = -1.699
pop_mean = ((samp_mean-t)*math.sqrt(n-1))/samp_std

In [33]:
pop_mean

209.91372418210295

here we will reject the null hypothesis.